http_connector.py

```diff
@@ -514,6 +545,173 @@ def _verb(self, param, method):
             data=body
         )
 
+    def _handle_get_file(self, param, method):
+
+        action_result = ActionResult(dict(param))
+        self.add_action_result(action_result)
+
+        hostname = param[HTTP_JSON_HOSTNAME]
+        file_path = param[HTTP_JSON_FILE_PATH]
+
+        endpoint = hostname + file_path
+        # /some/dir/file_name
+        file_name = file_path.split('/')[-1]
+
+        try:
+            ret_val, r = self._make_http_call(
+                action_result,
+                endpoint=endpoint,
+                method=method,
+                verify=param.get('verify_certificate', False)
+            )
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            return action_result.set_status(phantom.APP_ERROR, HTTP_SERVER_CONNECTION_ERROR_MESSAGE.format(error=err))
+
+        if (phantom.is_fail(ret_val)):
+            return action_result.get_status()
+
+        if (r.status_code == 200):
+            return self._save_file_to_vault(action_result, r, file_name)
+        else:
+            return action_result.set_status(phantom.APP_ERROR, HTTP_SERVER_CONNECTION_ERROR_MESSAGE.format(error=r.status_code))
+
+    def _handle_put_file(self, param, method):
+
+        action_result = ActionResult(dict(param))
+        self.add_action_result(action_result)
+
+        # fetching phantom vault details
+        try:
+            success, message, vault_meta_info = ph_rules.vault_info(vault_id=param[HTTP_JSON_VAULT_ID])
+            vault_meta_info = list(vault_meta_info)
```

Comment:
is `vault_meta_info` an empty iterator when `success == False` or is it `None`? if it's the latter then this line will raise an exception resulting in the actual `message` from the `vault_info` call to get swallowed